texar/torch/run/metric/summary.py

```diff
@@ -152,15 +153,25 @@ class LR(StreamingMetric[Any, float]):
 
     def __init__(self, optimizer: Optimizer, param_group: int = 0):
         super().__init__(pred_name=None)
-        self.optimizer = optimizer
+        self.optimizer = weakref.ref(optimizer)
```

Comment:
Why will the weak reference resolve the issue you mentioned? When will that `Optimizer` object be destroyed?

Reply:
Using a weak reference does not explicitly solve the problem, but the rationale here is to prevent storing unnecessary references to large objects because doing so might accidentally keep them alive beyond their lifetime.